src/docs/content/extras/canary.md

```diff
@@ -0,0 +1,242 @@
+---
+title: "Canary Deployment"
+date: 2020-03-10T13:29:23-06:00
+draft: false
+weight: 90
+tags:
+  - Canary
+  - Patching
+---
+
+{{% alert style="warning" icon="warning" %}}
+Note that this is **OPTIONAL**. This section is only applicable if you have completed [Istio service mesh]({{< ref "istio.md" >}}) section. 
+{{% /alert %}}
+
+## Introduction
+
+In this section we will demonstrate canary deployment use case with MuShop Application.
+
+One of the benefits of the Istio project is that it provides the control needed to deploy canary services. The idea behind canary deployment (or rollout) is to introduce a new version of a service by first testing it using a small percentage of user traffic, and then if all goes well, increase, possibly gradually in increments, the percentage while simultaneously phasing out the old version. If anything goes wrong along the way, we abort and rollback to the previous version. In its simplest form, the traffic sent to the canary version is a randomly selected percentage of requests, but in more sophisticated schemes it can be based on the region, user, or other properties of the request. Read more details [here](https://istio.io/latest/blog/2017/0.1-canary/) 
+
+We shall demonstrate the simplest canary deployment by using two versions of storefront microservice (storefront:original and storefront:beta) and would split traffic between the two. 
+
+- storefront:original would be the original page which does not have the reviews feature.
+- storefront:beta would have a new feature called "reviews" displayed on the UI which would let the user's provide review for their products. 
+
+Note: The builds storefront:original and storefront:beta naming conventions are used here for easier understanding. 
+Later during the procedure we will use the exact build version. 
+
+Lets look at the diagram which would explain the same
+
+![MuShop Canary Deployment](../images/mesh/mushop-canary.png)
+
+{{% alert style="warning" icon="warning" %}}
+The path between users and storefront has many layers ( DNS -> WAF -> LB -> INGRESS -> Router -> Storefront), The above figure shown is high level. Refer https://mushop.ateam.cloud/about.html for more details.
+{{% /alert %}}
+
+We would configure and run the storefront services in an Istio-enabled environment, with Envoy sidecars injected along side each service. We configure the Istio http routing by creating a VirtualService and Destination rules. Storefront images are available on the Oracle Cloud Infrastructure Registry. We would use them to create a kubernetes deployment.
+
+## Pre-Requisites:
+
+Deploy [Mushop]({{< ref "quickstart/kubernetes.md" >}})
+
+Download and install [Istio Service Mesh]({{< ref "istio.md" >}})
+
+## Deploy storefront beta
+ 
+ - Create a deployment with the below config and name it mushop-storefrontv2.
+   Note: We will use the same labels of storefront original.
+
+  ```shell--macos-linux
```

Comment:
I just realized that there are no windows equivalent snippets here. Are you able to provide those as well?